src/azure-cli-tests/azure/cli/tests/preparers.py

```diff
@@ -0,0 +1,173 @@
+        return _preparer_wrapper
+
+    @property
+    def moniker(self):
+        if not self.resource_moniker:
+            self.test_class_instance.test_resources_count += 1
+            self.resource_moniker = '{}{:06}'.format(self.name_prefix,
+                                                     self.test_class_instance.test_resources_count)
+        return self.resource_moniker
+
+    @property
+    def random_name(self):
+        if not self.resource_random_name:
+            self.resource_random_name = create_random_name(self.name_prefix, self.name_len)
+        return self.resource_random_name
+
+    def create_resource(self, name, **kwargs):
+        return {}
+
+    def remove_resource(self, name, **kwargs):
+        pass
+
+
+# TODO: replaced by GeneralNameReplacer
+class RecordingProcessorMixin(RecordingProcessor):
+    def process_request(self, request):
+        try:
+            request.uri = self.replace(request.uri)
+        except (KeyError, AttributeError):
+            pass
+
+        try:
+            request.body = self.replace(request.body)
+        except (KeyError, AttributeError, TypeError):
+            pass
+
+        return request
+
+    def process_response(self, response):
+        try:
+            response['body']['string'] = self.replace(response['body']['string'])
+        except (KeyError, AttributeError):
+            pass
+
+        try:
+            response['headers']['location'] = \
+                [self.replace(l) for l in response['headers']['location']]
+        except (KeyError, AttributeError):
+            pass
+
+        return response
+
+    def replace(self, original_value):
+        return original_value.replace(self.random_name, self.moniker)
+
+
+# Resource Group Preparer and its shorthand decorator
+
+class ResourceGroupPreparer(AbstractPreparer, RecordingProcessorMixin):
+    def __init__(self, parameter_name='resource_group',
+                 parameter_name_for_location='resource_group_location', location='westus'):
+        super(ResourceGroupPreparer, self).__init__('clitest.rg.', 90)
+        self.location = location
+        self.parameter_name = parameter_name
+        self.parameter_name_for_location = parameter_name_for_location
+
+    def create_resource(self, name, **kwargs):
+        template = 'az group create --location {} --name {} --tag use=az-test'
+        execute(template.format(self.location, name))
+        return {self.parameter_name: name, self.parameter_name_for_location: self.location}
+
+    def remove_resource(self, name, **kwargs):
+        execute('az group delete --name {} --yes --no-wait'.format(name))
+
+
+# Storage Account Preparer and its shorthand decorator
+
+class StorageAccountPreparer(AbstractPreparer, RecordingProcessorMixin):
+    def __init__(self, sku='Standard_LRS', location='westus',
+                 parameter_name='storage_account',
+                 resource_group_parameter_name='resource_group', skip_delete=True):
```

Comment:
Since resource group will be deleted afterwards, any resource created in it will be removed together. Therefore some of the resource like storage account doesn't have to be explicitly deleted.